components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/JDBCIdentityDataStore.java

```diff
@@ -187,7 +187,13 @@ public UserIdentityClaimsDO load(String userName, UserStoreManager userStoreMana
         PreparedStatement prepStmt = null;
         ResultSet results = null;
         try {
+
             int tenantId = userStoreManager.getTenantId();
+            if (!userStoreManager.isExistingUser(userName)) {
```

Comment:
Can't check isExistingUser in this level.
1. This is a DAO class to interact with JDBC IdentityDataStore. We should not have calls going out to user stores from here. That should happen at the service layer.
2. This method is called from many different places. In some of those places, e.g. like preDoAuthenticate we check for isUserExisting. In that case now isUserExisting is going to be called twice in successful cases, once in user event listener level and again here.